src/scan/postgres_table_reader.cpp

```diff
@@ -0,0 +1,177 @@
+#include "pgduckdb/scan/postgres_table_reader.hpp"
+#include "pgduckdb/pgduckdb_process_lock.hpp"
+
+extern "C" {
+#include "postgres.h"
+#include "miscadmin.h"
+#include "access/htup_details.h"
+#include "catalog/namespace.h"
+#include "catalog/pg_class.h"
+#include "optimizer/planmain.h"
+#include "optimizer/planner.h"
+#include "utils/builtins.h"
+#include "utils/regproc.h"
+#include "utils/rel.h"
+#include "utils/snapmgr.h"
+#include "utils/syscache.h"
+#include "tcop/tcopprot.h"
+#include "executor/execdesc.h"
+#include "executor/executor.h"
+#include "executor/execParallel.h"
+#include "executor/tqueue.h"
+#include "utils/wait_event.h"
+#include "access/xact.h"
+#include "pgduckdb/pgduckdb_guc.h"
+}
+
+#include <cmath>
+
+namespace pgduckdb {
+
+PostgresTableReader::PostgresTableReader(Cardinality cardinality, const char *table_scan_query)
+    : parallel_worker_readers(nullptr), nreaders(0), next_parallel_reader(0) {
+
+	std::lock_guard<std::mutex> lock(DuckdbProcessLock::GetLock());
+
+	pg_stack_base_t current_stack = set_stack_base();
```

Comment:
Postgresql asserts because of stack depth exceeds. With this we reset/set base of stack to current memory and returns to previous one once we exit function.